library/std/src/sys/windows/stdio.rs

```diff
@@ -50,7 +60,27 @@ fn is_console(handle: c::HANDLE) -> bool {
     unsafe { c::GetConsoleMode(handle, &mut mode) != 0 }
 }
 
-fn write(handle_id: c::DWORD, data: &[u8]) -> io::Result<usize> {
+// Simple reimplementation of std::str::utf8_char_width() which is feature-gated
```

Comment:
any reason not to dog-food that one if it already exists? std is allowed to use its own unstable features.

Reply:
@the8472 To be honest, I just did not know under what circumstances this is allowed and how to do it. 
This my first time working on the Rust codebase.

Reply:
You can add `#![feature(str_internals)]` to the crate root.